.travis.yml

```diff
@@ -1,41 +1,147 @@
-  - TAR_EXCLUDES="--exclude=docker --exclude=features --exclude=run_tests.sh --exclude=*node_modules"
+# ------------------------
+# Jobs configuration
+# ------------------------
+jobs:
+  include:
+    # ----------------------------
+    # Deadlinks Tests
+    # ----------------------------
+    - name: Dead link check
+      stage: Tests
+      language: node_js
+      node_js: "10"
+      cache:
+        directories:
+          - "$HOME/.gem/specs"
+          - ".doc/framework/node_modules"
+      before_script:
+        - npm run doc-prepare
+        - npm run --prefix doc/framework clone-repos
+      script:
+        - gem install typhoeus
+        - HYDRA_MAX_CONCURRENCY=20 npm run --prefix doc/framework dead-links
 
-before_install:
-  - sudo sysctl -w vm.max_map_count=262144
-  # Build images
-  - docker login -u aschen -p $DOCKER_PASSWORD
-  - for dir in $(\ls -1d */docker); do sh -c "cd $dir && bash build_images.sh && bash push_images.sh"; done
+    # ----------------------------
+    # Archive deployment
+    # ----------------------------
+    - <<: *deploy-base
+      name: Deploy Archives to dl.kuzzle.io
+      if: branch = master
+      stage: Deployments
 
-addons:
-  apt:
-    packages:
-      - ruby
-      - awscli
+      install:
+        - pip install awscli --upgrade --user
+      before_install:
+        - sudo sysctl -w vm.max_map_count=262144
+        # Build images
+        - docker login -u aschen -p $DOCKER_PASSWORD
+        - |
+          for dir in $(\ls -1d */docker)
+          do
+            cd $dir
+            bash build_images.sh
+            bash push_images.sh
+          done
+      script:
+        - mkdir build
+        # Run tests
+        - |
+          for dir in */
+          do
+            base=$(basename "$dir")
+            if [ $base != 'samples' ] && [ $base != 'build' ]
+            then
+              cd ${base}
+              ./run_tests.sh
+              cd -
+            fi
+          done
+      after_success:
+        # Create how-to.tar.gz
+        - |
+          for dir in */
+          do
+            base=$(basename "$dir")
+            if [ $base != 'samples' ] && [ $base != 'build' ]
+            then
+              tar $TAR_EXCLUDES -cvzf build/${base}.tar.gz ${base}
+            fi
+          done
+        - aws s3 cp build s3://$S3_BUCKET/how-to/ --recursive --region us-west-2
+    
+    # ----------------------------
+    # Prod deployment
+    # ----------------------------
+    - <<: *deploy-base
+      <<: *doc-deploy-content
+      name: Deploy docs.kuzzle.io
+      if: branch = master
+      stage: Deployments
+      env:
+        - *deploy-base-env
+        - S3_BUCKET=docs.kuzzle.io
+        - CLOUDFRONT_DISTRIBUTION_ID=E3D6RP0POLCJMM
+  
+    # ----------------------------
+    # Dev deployment
+    # ----------------------------
+    - <<: *deploy-base
+      <<: *doc-deploy-content
+      name: Deploy next-docs.kuzzle.io
+      if: branch = develop
```

Comment:
Do we have a develop branch ? If no, we need to deploy both `docs.kuzzle.io` and `next-docs.kuzzle.io` when pushing on `master`

Reply:
He creates one :wink: